packages/dd-trace/src/util.js

```diff
@@ -87,12 +87,83 @@ function generatePointerHash (components) {
   return hash.substring(0, 32)
 }
 
+/**
+ * Encodes a DynamoDB attribute value to Buffer for span pointer hashing.
+ * @param {Object} valueObject - DynamoDB value in AWS format ({ S: string } or { N: string } or { B: Buffer })
+ * @returns {Buffer} Encoded value as Buffer, or empty Buffer if invalid input.
+ *
+ * @example
+ * encodeValue({ S: "user123" }) -> Buffer("user123")
+ * encodeValue({ N: "42" }) -> Buffer("42")
+ * encodeValue({ B: Buffer([1, 2, 3]) }) -> Buffer([1, 2, 3])
+ */
+function encodeValue (valueObject) {
+  if (!valueObject) {
+    return Buffer.from('')
+  }
+
+  try {
+    const type = Object.keys(valueObject)[0]
+    const value = valueObject[type]
+
+    switch (type) {
+      case 'S':
+        return Buffer.from(value)
+      case 'N':
+        return Buffer.from(value.toString())
+      case 'B':
+        return Buffer.isBuffer(value) ? value : Buffer.from(value)
+      default:
+        return Buffer.from('')
+    }
+  } catch (err) {
+    return Buffer.from('')
+  }
+}
+
+/**
+ * Extracts and encodes primary key values from a DynamoDB item.
+ * Handles tables with single-key and two-key scenarios.
+ *
+ * @param {Set<string>|Object} keySet - Set of key names or object of key names/value pairs.
+ * @param {Object} keyValuePairs - Object containing key/value pairs.
+ * @returns {Array|undefined} [key1Name, key1Value, key2Name, key2Value], or undefined if invalid input.
+ *                            key2 entries are empty strings in the single-key case.
+ * @example
+ * extractPrimaryKeys(new Set(['userId']), {userId: {S: "user123"}})
+ * // Returns ["userId", Buffer("user123"), "", ""]
+ * extractPrimaryKeys(new Set(['userId', 'timestamp']), {userId: {S: "user123"}, timestamp: {N: "1234}})
+ * // Returns ["timestamp", Buffer.from("1234"), "userId", Buffer.from("user123")]
+ */
+const extractPrimaryKeys = (keySet, keyValuePairs) => {
+  const keyNames = keySet instanceof Set
+    ? Array.from(keySet)
+    : Object.keys(keySet)
+  if (keyNames.length === 0) {
+    return
+  }
+
+  if (keyNames.length === 1) {
+    return [keyNames[0], encodeValue(keyValuePairs[keyNames[0]]), '', '']
+  } else {
+    const [key1, key2] = keyNames.sort()
+    return [
+      key1,
+      encodeValue(keyValuePairs[key1]),
+      key2,
+      encodeValue(keyValuePairs[key2])
+    ]
+  }
+}
+
 module.exports = {
   isTrue,
   isFalse,
   isError,
   globMatch,
   calculateDDBasePath,
   hasOwn,
-  generatePointerHash
+  generatePointerHash,
+  encodeValue,
```

Comment:
I had to export `encodeValue` to test it, but it's not used externally. Is this fine or is there something better practice I can do?